projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts

```diff
@@ -0,0 +1,168 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { AddressMessageEnum, AddressModel } from '../../models';
+import { AddressOption } from '../../models/address-option.model';
+import { AddressService } from '../../services/address/address.service';
+
+@Component({
+  selector: 'xuilib-write-address-field',
+  styleUrls: ['./write-address.component.scss'],
+  templateUrl: './write-address.component.html'
+})
+export class WriteAddressFieldComponent implements OnInit, OnChanges {
+
+  @Input()
+  public formGroup: FormGroup;
+  @Input()
+  public internationalMode = false;
+  @Input()
+  public isInternational: boolean;
+  @Input()
+  public submissionAttempted = false;
+  @Input()
+  public startedInternational: boolean;
+
+  @Output() public postcodeOptionSelected = new EventEmitter<void>();
+  @Output() public internationalModeStart = new EventEmitter<void>();
+  @Output() public ukAddressOptionSelected = new EventEmitter<boolean>();
+
+  public addressesService: AddressService;
+
+  public addressField: AddressModel = {
+    addressLine1: '',
+    addressLine2: '',
+    addressLine3: '',
+    postCode: '',
+    postTown: '',
+    country: '',
+    county: ''
+  };
+
+  public optionErrorsPresent = false;
+
+  public addressChosen = false;
+
+  public addressFormGroup: FormGroup;
+
+  public addressOptions: AddressOption[];
+
+  public missingPostcode = false;
+
+  public optionErrorMessage = AddressMessageEnum.NO_OPTION_SELECTED;
+
+  constructor(addressesService: AddressService) {
```

Comment:
@connorpgpmcelroy - should be a private read-only property in the constructor?

Reply:
this was due to porting from toolkit, I will change though